core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -1852,12 +1857,19 @@ allocate_iovec_app_buffer(wasm_module_inst_t module_inst,
     return __WASI_ESUCCESS;
 }
 
+static inline size_t
+min(size_t a, size_t b)
+{
+    return a > b ? b : a;
+}
+
 static wasi_errno_t
 copy_buffer_to_iovec_app(wasm_module_inst_t module_inst, uint8 *buf_begin,
                          uint32 buf_size, iovec_app_t *data, uint32 data_len)
```

Comment:
I think for clarity we should update the signature of this function:
 * `buf_size` doesn't really represent a size of the buffer pointed by `buf_begin`; it represents a number of bytes to copy (at least based on the call in @L1924
 * we should probably pass both values to the function - size of the buffer (`buf_size`) and a number of bytes to copy (e.g. `size_to_copy`). If `buf_size < size_to_copy`, I think the function should return `EINVAL`, instead of silently swallowing the problem. 

Once we do that, later on in the function we know for sure that the buffer is enough to copy all the data, but I agree we should only copy the `size_to_copy` amount for efficiency.

Reply:
Will address thanks for suggestions